tst/EnergyPlus/unit/extendedHI.unit.cc

```diff
@@ -0,0 +1,734 @@
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+// EnergyPlus Headers
+#include "Fixtures/EnergyPlusFixture.hh"
+#include <EnergyPlus/Data/EnergyPlusData.hh>
+#include <EnergyPlus/HVACSystemRootFindingAlgorithm.hh>
+#include <EnergyPlus/extendedHI.hh>
+
+using namespace EnergyPlus;
+
+TEST_F(EnergyPlusFixture, extendedHI_pvstar)
+{
+    Real64 tol = 1e-8;
+    std::vector<Real64> T_values = {200, 210, 220, 230, 240, 250, 260, 270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370};
+    std::vector<Real64> result = {0.16315953136381425,
+                                  0.7045737646159884,
+                                  2.6639212639290717,
+                                  8.972721335550697,
+                                  27.315394188834492,
+                                  76.0747215114313,
+                                  195.83100375810935,
+                                  470.0335224802147,
+                                  991.9254222637857,
+                                  1920.6801555359402,
+                                  3538.9408236895683,
+                                  6235.887915937109,
+                                  10554.04916628072,
+                                  17222.314773780665,
+                                  27187.715714871047,
+                                  41643.766112228834,
+                                  62053.2640569111,
+                                  90163.72448626769};
```

Comment:
I'm inclined to say all the values in this unit test are _waaay_ too precise.  We cannot know these values so precisely, so to test against them is futile.  I see you have tolerance set to e-8, which is pretty tight here.  But there are other ones where tolerance is more like e-4, yet the values you compare against have like 12+ decimals.  It's not the end of the world, it's just a lot of numbers here :)

Reply:
I will unify the tolerance and the values here.

These are output from this python file. I was trying to ideally match their result. 
[heatindex.py.zip](https://github.com/user-attachments/files/16146813/heatindex.py.zip)

The e-4 part is due to the difference in the root solvers between EnergyPlus and the source python file  (when EnergyPlus uses the bisection one as well).
The following is a illustration of the difference in a somewhat extreme case
![image](https://github.com/NREL/EnergyPlus/assets/13421161/6697fd07-a6b1-455d-8e89-321536043d2f)
Here is the difference in the code
<img width="1216" alt="image" src="https://github.com/NREL/EnergyPlus/assets/13421161/d4b9a017-49cf-453a-be3c-449580395849">